components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/SchemaValidator.java

```diff
@@ -0,0 +1,610 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.security;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.AbstractHandler;
+import org.apache.synapse.transport.passthru.util.RelayUtils;
+import org.json.JSONObject;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.gateway.APIMgtGatewayConstants;
+import org.wso2.carbon.apimgt.gateway.threatprotection.utils.ThreatProtectorConstants;
+import org.wso2.carbon.apimgt.gateway.utils.GatewayUtils;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Iterator;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.jayway.jsonpath.JsonPath;
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.AxisFault;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.commons.json.JsonUtil;
+import org.apache.synapse.config.Entry;
+import org.everit.json.schema.Schema;
+import org.everit.json.schema.ValidationException;
+import org.everit.json.schema.loader.SchemaLoader;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+
+/**
+ * This SchemaValidator handler validates the request/response messages against schema defined in the swagger.
+ */
+public class SchemaValidator extends AbstractHandler {
+
+    private static final Log logger = LogFactory.getLog(SchemaValidator.class);
+    private String uuid;
+    private String swagger = null;
+    private JsonNode rootNode;
+    private String requestMethod;
+    private String schemaContent = null;
+
+    public void setLocalentry(String localEntry) {
+        this.uuid = localEntry;
+    }
+
+    /**
+     * Handle the API request message validation.
+     *
+     * @param messageContext Context of the message to be validated
+     * @return Continue true the handler chain
+     */
+    @Override
+    public boolean handleRequest(MessageContext messageContext) {
+        if (logger.isDebugEnabled()) {
+            logger.info("Validating the API request Body content..");
```

Comment:
This should a debug log
```suggestion
            logger.debug("Validating the API request Body content..");
```